src/components/AdminDashboard/forms/AdminEdit.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import { Form, Input, Select, DatePicker, Button } from 'antd';
+import moment from 'moment';
+import useOktaAxios from '../../../hooks/useOktaAxios';
+
+import './AdminEdit.less';
+
+const { Option } = Select;
+
+function AdminEdit({ initialValues, cancel, cleanup }) {
+  const [form] = Form.useForm();
+
+  const oktaAxios = useOktaAxios();
+
+  const handleSubmit = vals => {
+    let formattedDate;
+
+    // Making sure that the date is a moment
+    if (vals.incident_date === null) {
+      formattedDate = moment(initialValues.incident_date);
+    } else {
+      formattedDate = vals.incident_date;
+    }
+    // Putting the date in the correct format
+    formattedDate = formattedDate.format('YYYY-MM-DD') + 'T00:00:00.000Z';
+
+    // Removing duplicate tags, ex: "asdf " and "asdf"
+    let formattedTags = new Set(vals.tags.map(tag => tag.trim()));
+
+    const finalVals = {
+      ...initialValues,
+      ...vals,
+      incident_date: formattedDate,
+      tags: [...formattedTags],
+    };
+
+    oktaAxios
+      .put('/dashboard/incidents', [{ ...finalVals }])
+      .then(res => {
+        window.location.reload();
```

Comment:
Yeah, I've noticed this pattern in a few places, where a subcomponent far down on the AdminDashboard makes a change to the data and reloads the whole page. Thanks for pointing it out (and for pointing out the the finally block is pointless).

Reply:
Yeah I didn't really feel like it was in the scope of this PR, but also reloading the whole SPA to update state is not a thing I want to be doing.